store/models.py

```diff
@@ -0,0 +1,85 @@
+from django.db import models
+from wagtail.core.fields import RichTextField
+from wagtail.core.models import Page
+from wagtail.admin.edit_handlers import FieldPanel
+from wagtail.images.edit_handlers import ImageChooserPanel
+
+from cart.forms import CartAddProductForm
+
+
+class StoreIndexPage(Page):
+    intro = RichTextField(blank=True)
+
+    content_panels = Page.content_panels + [
+        FieldPanel("intro"),
+    ]
+
+    subpage_types = [
+        "store.CategoryIndexPage",
+        "store.ProductIndexPage",
+    ]
+
+    max_count = 1
+
+    def get_context(self, request, *args, **kwargs):
+        context = super().get_context(request)
+
+        context["products"] = Product.objects.all()
+
+        return context
+
+
+class CategoryIndexPage(Page):
+
+    max_count = 1
+
+    subpage_types = [
+        "store.Category",
+    ]
+
+
+class Category(Page):
+
+    subpage_types = []
+
+    class Meta:
+        verbose_name = "category"
+        verbose_name_plural = "categories"
+
+
+class ProductIndexPage(Page):
+
+    max_count = 1
+
+    subpage_types = [
+        "store.Product",
+    ]
+
+
+class Product(Page):
```

Comment:
Book products will need `author` field. See if it is possible to create models that inherit from this, in a generic way that can be used by the cart.